packages/docs-site/src/content/docs/docs/guides/multi-env.mdx

```diff
@@ -3,79 +3,82 @@ title: Multi-environment configuration
+
+To use `switchBy`, we select another item by name to use as the switch condition, and provide a key-value object to define the different branches that might be selected. The keys are the possible expected values, with `_default` being a special reserved key that is selected if no match is found.
+
+Note that each branch can be a static value, a function, or a resolver - just like setting the value itself. This leads to some powerful composition capabilities.
 
 ```typescript
-import { switchByNodeEnv } from 'dmno';
+import { NodeEnvType, switchBy } from 'dmno';
 
 export default defineDmnoService({
-  pick: [
-    'NODE_ENV',
-  ],
   schema: {
-    OTHER_ITEM: {},
-    MY_CONFIG_ITEM: {
-      value: switchByNodeEnv({
-        _default: 'default value',
-        development: 'development value',
+    NODE_ENV: NodeEnvType,
+    TOGGLED_ITEM: {
+      value: switchBy('NODE_ENV', {
+        _default: 'default/development value',
         staging: 'staging value',
+        test: 'test value',
         production: () => {
           return DMNO_CONFIG.OTHER_ITEM ? 'production value' : 'production value 2';
         },
       }),
     },
+    SOME_API_KEY: {
+      sensitive: true,
+      value: switchBy('NODE_ENV', {
+        _default: devSecretsVault.item(),
+        production: prodSecretsVault.item(),
+      },
+    },
   },
 });
 ```
 
-:::tip
-These don't need to be static values, they can be any valid configuration item, including references to secrets stored in a vault.
-:::
-
-You can define as many configuration items as you need using the `switchByNodeEnv` method.
-
-## Environment variables
-
-To set the environment for your application, you can use the `NODE_ENV` environment variable. This variable is commonly set to `development`, `staging`, or `production` depending on the environment.
-
-For example, to set the environment to `production` and do a production build, you could run:
-
-```bash
-NODE_ENV=production pnpm exec dmno run -- my-build-command-goes-here
-```
-
-This is especially useful when the value of `NODE_ENV` might not be what you would expect (we're looking at you [Netlify](https://docs.netlify.com/configure-builds/manage-dependencies/#node-js-environment)). For example, when running a build on a CI server, you might want to set the environment to `production` to ensure that the correct configuration values are used.
+## Which environment flag should you use?
 
-## Other environments
+While `NODE_ENV` is a common env flag that you are probably familiar with, we do not recommend using it! Historically, some 3rd party modules have altered their own behaviour based on this flag, so it's best to avoid it and leave `NODE_ENV=production`, especially in a staging environment where we want to run production-like code. It is common that other platforms inject their own env flag for the same reason.
```

Comment:
```suggestion
While `NODE_ENV` is a common env flag that you are probably familiar with, **we do not recommend using it**. Historically, some 3rd party modules have altered their own behaviour based on this flag, so it's best to avoid it and leave `NODE_ENV=production`, especially in a staging environment where we want to run production-like code. It is common that other platforms inject their own env flag for the same reason.
```